src/libmongoc/src/mongoc/mongoc-client-session-private.h

```diff
@@ -83,6 +84,9 @@ struct _mongoc_client_session_t {
    uint32_t client_generation;
    uint32_t server_id;
    bson_t *recovery_token;
+   uint32_t snapshot_time_timestamp;
+   uint32_t snapshot_time_increment;
+   bool snapshot_time_set;
```

Comment:
This was my best guess at storing an optional timestamp 🤷 . It looks like timestamps are often stored as their `timestamp` and `increment` values (see `operation_timestamp` and `operation_increment` above), but I needed a `bool` as well (`snapshot_time_set`) to know when the snapshot time was set. I tried to use two `uint32_t *`  but the pointers lost the underlying value as I passed them around. Let me know what you think of this approach (I do something similar for `at_cluster_time` on `read_concern`).

Reply:
Ah yeah that'd be useful here. These `snapshot_time` fields aren't exposed API, but the `at_cluster_time` fields are, so it might be worthwhile for me to implement `mongoc_optional_t` for timestamps if @kevinAlbs agrees.

Reply:
No longer necessary since `at_cluster_time` fields are gone.